### ISSUE TITLE
OrbitControls: Speed and sensitivity is too high

### ISSUE THREAD

**Issue by DarceyLloyd** [mrdoob/three.js#9577]
##### Speeds are insane / Crazy sensitive, inertia, unable to get correct settings.

Quite possibly the most essential Camera Controller that comes with ThreeJS (OrbitControls), which directly shows off the capabilities of threejs is having some issues. If you take a look at TrackballController, a great camera controller however it's not very useful in a product as orientation is lost but its speed of rotation and inertial (dampening) values are good and give a good feel, however ObrbitControls doesn't come close. Not once through the life cycle of papervision and away3d did I ever apply trackballcontrollers style of orientation to a commercial job, dev tests maybe a few times but not many, it was always OrbitControls functionality.
##### Speed issue, mouse drag distance, sensitivity

An example of this can be seen here http://threejs.org/examples/misc_controls_orbit.html just drag the mouse a little bit faster than slow and your about to rotate around that scene over 5000 times, and there are many more like it. I've noticed many a dev & example using the TrackballController even though it's orientation is lost so easily and should have been OrbitControls. However as OrbitControls is so extremely sensitive and some strange inertia / dampening capabilities it can't match it.

I've been trying to find an alternative to OrbitControls for some time, something like BABYLON.ArcRotateCamera or Away3Ds HoverCamera but haven't seen anything. Most other examples end up using mouse events to rotate the object rather than the camera around the object while looking at it.

I've been trying to get THREE.OrbitControls to simulate BABYLON.ArcRotateCamera but the closest I can get is with the following code but it's just not right. The close I get to the correct inertia the more sensitive the drag distance gets and adjusting the speed then throws off the dampening factor.

```
        this.cameraControls = new THREE.OrbitControls(this.camera, this.renderer.domElement);
        this.cameraControls.target.set(0, 0, 0);
        this.cameraControls.enableDamping = true;
        this.cameraControls.dampingFactor = 0.04;
        this.cameraControls.rotateSpeed = 0.07;
```

Maybe time for an update/revision?

ThreeJS awesome by the way, very nice.
##### Issues
1. Speed (way too fast in relation to mouse distance - desktop)
2. dampening (inertia) impossible to simulate dampening and rotation speeds of TrackBallControls or the BabylonJS equivalent, you always end up with something close but not correct.
##### Three.js version (many many versions)
- [ ] Dev
- [x] r60 to r80
- [ ] ...
##### Browser
- [x] All of them
##### OS
- [x] All of them

**Comment by WestLangley** [comment]
>An example of this can be seen here http://threejs.org/examples/misc_controls_orbit.html just drag the mouse a little bit faster than slow and your about to rotate around that scene over 5000 times

Really? I do not see that behavior.

**Comment by DarceyLloyd** [comment]
Video of what I get: https://youtu.be/JHp2ZEjG3b0

**Comment by WestLangley** [comment]
Your free parameters:
```javascript
controls.dampingFactor = 0.1; // friction
controls.rotateSpeed = 0.1; // mouse sensitivity
```
The behavior will also be affected by the mouse sensitivity setting in your OS.

**Comment by DarceyLloyd** [comment]
Understood that mouse sensitivity is affected by mouse DPI settings and OS settings (speed & acceleration), I don't have it set at fast or slow, usually left at defaults and I don't change DPI as the Razer software crashes if I try to do that.

I've tried those settings, I've even gone to the point of bolting sliders to the values to play with them, best I can come up with is:

```
this.cameraControls.dampingFactor = 0.07;
this.cameraControls.rotateSpeed = 0.04;
```

Video of this for you: https://youtu.be/NOT0oOQE8_M

You just wont get it the same as TrackBallControls or ArcRotateCamera.

Is there an easing curve (out)? Applied to the dampening?

**Comment by vudn54** [comment]
I want to add orbit camera to loader mtl but not success.  

	</head>

	<body>
		<div id="info">
		<a href="http://threejs.org" target="_blank">three.js</a> - OBJLoader + MTLLoader
		</div>
		<script src="../build/three.js"></script>
		<script src="js/loaders/DDSLoader.js"></script>
		<script src="js/loaders/MTLLoader.js"></script>
		<script src="js/loaders/OBJLoader.js"></script>
<script src="OrbitControls.js"></script>
		<script src="js/Detector.js"></script>
		<script src="js/libs/stats.min.js"></script>
		<script>
			var container;
var camera, controls, scene, renderer;
var lighting, ambient, keyLight, fillLight, backLight;
var windowHalfX = window.innerWidth / 2;
var windowHalfY = window.innerHeight / 2;
			init();
			animate();
			function init() {
				container = document.createElement( 'div' );
				document.body.appendChild( container );
				camera = new THREE.PerspectiveCamera(45, window.innerWidth / window.innerHeight, 1, 1000);
camera.position.z = 3;
				// scene
				scene = new THREE.Scene();
ambient = new THREE.AmbientLight(0xffffff, 1.0);
scene.add(ambient);
				var directionalLight = new THREE.DirectionalLight( 0xffeedd );
				directionalLight.position.set( 0, 0, 1 ).normalize();
				scene.add( directionalLight );
				// model
				var onProgress = function ( xhr ) {
					if ( xhr.lengthComputable ) {
						var percentComplete = xhr.loaded / xhr.total * 100;
						console.log( Math.round(percentComplete, 2) + '% downloaded' );
					}
				};
				var onError = function ( xhr ) { };
				THREE.Loader.Handlers.add( /\.dds$/i, new THREE.DDSLoader() );
				var mtlLoader = new THREE.MTLLoader();
				mtlLoader.setPath( 'obj/obj/' );
				mtlLoader.load( 'lade.mtl', function( materials ) {
					materials.preload();
					var objLoader = new THREE.OBJLoader();
					objLoader.setMaterials( materials );
					objLoader.setPath( 'obj/obj/' );
					objLoader.load( 'lade.obj', function ( object ) {
						object.position.y = - 95;
						scene.add( object );
					}, onProgress, onError );
				});
				//
				renderer = new THREE.WebGLRenderer();
				renderer.setPixelRatio( window.devicePixelRatio );
				renderer.setSize( window.innerWidth, window.innerHeight );
				container.appendChild( renderer.domElement );
				document.addEventListener( 'mousemove', onDocumentMouseMove, false );
				//
				window.addEventListener( 'resize', onWindowResize, false );
			}
			function onWindowResize() {
				windowHalfX = window.innerWidth / 2;
				windowHalfY = window.innerHeight / 2;
				camera.aspect = window.innerWidth / window.innerHeight;
				camera.updateProjectionMatrix();
				renderer.setSize( window.innerWidth, window.innerHeight );
			}
this.cameraControls = new THREE.OrbitControls(this.camera, this.renderer.domElement);
        this.cameraControls.target.set(0, 0, 0);
        this.cameraControls.enableDamping = true;
        this.cameraControls.dampingFactor = 0.04;
        this.cameraControls.rotateSpeed = 0.07;
		</script>
	}
	</body>
</html>

**Comment by DarceyLloyd** [comment]
OrbitCamera is not part of that process it's the camera, it should be setup as with the renderer and scene.

So you have scene, camera, renderer and cameraController = new OribtControls(xxxxx) done before you start loading/setting up lights, mesh's and textures. A Typical sequentially setup process goes like:

setupEngine(); // scene, camera, renderer and cameraController
setupLights();
setupMaterials();
setupScene(); // Objects / models / effects
setupEventListeners(); // Resize event listener etc
render(); // render loop

View source code on for example:
http://threejs.org/examples/misc_controls_orbit.html

**Comment by DarceyLloyd** [comment]
You may be able to load a camera in from a belnder export but I've always done it the way I just detailed.

**Comment by mrdoob** [comment]
> I've been trying to get THREE.OrbitControls to simulate BABYLON.ArcRotateCamera but the closest I can get is with the following code but it's just not right. The close I get to the correct inertia the more sensitive the drag distance gets and adjusting the speed then throws off the dampening factor.

Lets ignore dampingFactor for now (as in enableDamping=false).
What would be the ideal default rotateSpeed?

**Comment by DarceyLloyd** [comment]
The code I ended up using was this:
```
        this.cameraControls = new THREE.OrbitControls(this.camera, this.renderer.domElement);
        this.cameraControls.target.set(0, 0, 0);
        this.cameraControls.enableDamping = true;
        this.cameraControls.dampingFactor = 0.07;
        this.cameraControls.rotateSpeed = 0.04;
```

But for reasonable rotation/movement speeds I would just think of what is practical for client applications. So if I were to do a panoramic or look around a 3D object or something like that you would be looking at something more along the lines of the defaults in Away3Ds HoverCamera and with BabylonJS ArcRotateCamera, even if making a simple game look around is too fast and extremely sensitive. The TrackballControls are close and has good inertia / dampening values.

BabylonJS ArcRotateCamera: https://youtu.be/4FfnF42t_zc
Away3D HoverCamera: https://youtu.be/f7EnfHRGzF4

https://www.allforthecode.co.uk/old/aftc/forum/user/modules/forum/article.php?index=4&subindex=2&aid=333

**Comment by Mugen87** [comment]
I guess we can close this issue. It has been discussed several times that the sensitivity of the controls depends on various factors. `OrbitControls` provides a comprehensive set of parameters so users can adjust the behavior.

**Comment by DarceyLloyd** [comment]
I'm not sure I've not checked it in a long time, whether it has been updated or not. But speed should be based on velocity of mouse while down / touch drag and distance.  Inertia should have a more sensible fall off curve (power2), and with reasonable values. eg: https://greensock.com/ease-visualizer

I ended up using alternate solutions for when I posted this originally.

**Comment by mrdoob** [comment]
Yeah, I also noticed this issue recently https://github.com/mrdoob/three.js/pull/13706#issuecomment-377087386

I think we should take `window.devicePixelRatio` into account in these controls.

For example, I think [this line](https://github.com/mrdoob/three.js/blob/dev/examples/js/controls/OrbitControls.js#L458) should be:

```js
rotateEnd.set( event.clientX, event.clientY ).divideScalar( window.devicePixelRatio );
```

/cc @WestLangley

**Comment by samipe** [comment]
I think there is a bug that causes much faster orbiting speeds if dampening is enabled, just made a PR about this:
https://github.com/mrdoob/three.js/pull/17059

**Comment by DarceyLloyd** [comment]
If file size is not an issue, use Babylon much smoother camera capabilities out of the box (go -1 version for smaller again) alternatively you can spend a lot of time finding the examples for the old school flash approach where they have mouse event listeners for down, drag, up and vars for each with up and down coords calc and then tailor to a more reasonable speed and far far far better inertia to a standing state.

**Comment by WestLangley** [comment]
>WestLangley closed this in #17060 

I did??  It must be automatic, then...

In any event, a new post can be opened if there are additional issues.

**Comment by Mugen87** [comment]
> I did?? It must be automatic, then...

When you use words like "fix" or "solve" before an issue hash,  the PR will automatically close the issue when it gets merged. Notice the underlined word in your comment.

![image](https://user-images.githubusercontent.com/12612165/61589385-5e370e80-aba9-11e9-951b-fd9e6b907ec5.png)
